esp_idf_monitor/base/logger.py

```diff
@@ -91,16 +92,42 @@ def print(self, string, console_printer=None):  # noqa: E999
         if console_printer is None:
             console_printer = self.console.write_bytes
 
-        if self.timestamps and (self._output_enabled or self._log_file):
+        if string and self.timestamps and (self._output_enabled or self._log_file):
             t = datetime.datetime.now().strftime(self.timestamp_format)
+
             # "string" is not guaranteed to be a full line. Timestamps should be only at the beginning of lines.
             if isinstance(string, type(u'')):
                 search_patt = '\n'
-                replacement = '\n' + t + ' '
+                replacement = t + ' '
             else:
                 search_patt = b'\n'  # type: ignore
-                replacement = b'\n' + t.encode('ascii') + b' '  # type: ignore
+                replacement = t.encode('ascii') + b' '  # type: ignore
```

Comment:
Can you please make the names of the variables more clear? I know they were already like that, but now when they are used multiple times it is easier to get lost. IMO something like `new_line_char` and `line_prefix` would make more sense.